### DISCUSSION TITLE
What does NotFound do?

### DISCUSSION THREAD

**Discussion by Miljankg** [ardalis/GuardClauses#172] [Q&A]
Hello,

The API contains NotFound clause. However, the implementation of it only checks for whether the Key and Input are null. What would the use case for this clause be?

The statement from the tests `Guard.Against.NotFound(1, Guid.Empty, "guid");` will not throw, and I was not able to figure out where would I apply such statement.

Thanks in advance.

**Comment by AkhmedovEhson** [comment]
Hi, there !  
♻️  `Guard.Against.NotFound( KEY, VALUE, PARAMNAME )`

🔑 KEY : This is id of queried object 
📓  VALUE: This is input
🖊️  PARAMNAME : This is queried object's name 

Example: 
int productId = 1;
var product = context.Products.Where(o => o.Id == productId).FirstOrDefault();
Guard.Against.NotFound("id",product,"product"); // throws `NotFoundException` if product is null

**Reply by ardalis** [reply]
Personally I would use a Result not a guard/exception for this kind of scenario.

**Reply by AkhmedovEhson** [reply]
Heey ardalis :)
<br/>Actually I am using Exception/Filters. `GuardClauses` is more appropriate.
